app/src/components/ScrabbleHand.js

```diff
@@ -1,14 +1,33 @@
 import ScrabbleTile from './ScrabbleTile'
+import './ScrabbleHand.scss'
+import { useContext } from 'react';
+import { GameContext } from '../contexts/GameContext';
 
 export default function ScrabbleHand({ arrOfPlayersCurrentLetters }) {
 
+    const { _setCurrentTile, currentTile } = useContext(GameContext);
+
+    async function setCurrentTile(value, score, id) {
+        if (!currentTile || currentTile.id !== id) {
+            _setCurrentTile({ value, score, id });
+        } else if (currentTile.id === id) {
+            _setCurrentTile(null);
+        }
+    }
+
     return (
         <>
             <div className='container border border-warning p-0 text-center'>
                 {arrOfPlayersCurrentLetters.length > 0 ? (
                     <>
-                        {arrOfPlayersCurrentLetters.map((tile, index) => (
-                            <ScrabbleTile key={index} value={tile.value} score={tile.score} />
+                        {arrOfPlayersCurrentLetters.map((tile) => (
+                            <ScrabbleTile
+                                id={tile.id}
+                                key={tile.id}
+                                classNameForSelecting={currentTile?.id === tile.id ? 'selected' : ''}
```

Comment:
Ez a változónév, hogy `classNameForSelecting` szintén túl hosszú, túl specifikus. A `ScrabbleTile` "nem tud semmit" arról, hogy ezt most selectingre használod, vagy mire. Ő csak annyit csinál, hogy amit itt megkap, azt hozzáadja a classokhoz. Ez alapján lehetne akár `extraClasses` is a prop neve, ami sokkal rövidebb, elegánsabb, és multifunckiósabb